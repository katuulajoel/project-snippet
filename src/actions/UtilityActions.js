import axios from "axios";
import {
  composeFormData,
  ENDPOINT_INVITE_REQUEST,
  ENDPOINT_CONTACT_REQUEST,
  ENDPOINT_MEDIUM,
  ENDPOINT_MIGRATE,
} from "./utils/api";
import {
  CLEAR_VALIDATIONS,
  SEND_CONTACT_REQUEST_START,
  SEND_CONTACT_REQUEST_SUCCESS,
  SEND_CONTACT_REQUEST_FAILED,
  SEND_INVITE_REQUEST_START,
  SEND_INVITE_REQUEST_SUCCESS,
  SEND_INVITE_REQUEST_FAILED,
  GET_MEDIUM_POSTS_START,
  GET_MEDIUM_POSTS_SUCCESS,
  GET_MEDIUM_POSTS_FAILED,
  FIND_REPLACEMENT_START,
  FIND_REPLACEMENT_SUCCESS,
  FIND_REPLACEMENT_FAILED,
  TOGGLE_RIGHT_NAV,
} from "./utils/ActionTypes";

export const clearValidations = () => {
  return {
    type: CLEAR_VALIDATIONS,
  };
};

export const sendContactRequest = (data) => {
  return (dispatch) => {
    dispatch(sendContactRequestStart(data));
    axios
      .post(ENDPOINT_CONTACT_REQUEST, data)
      .then((response) => {
        dispatch(sendContactRequestSuccess(response.data));
      })
      .catch((error) => {
        dispatch(sendContactRequestFailed(error.response ? error.response.data : null));
      });
  };
};

export const sendContactRequestStart = (data) => {
  return {
    type: SEND_CONTACT_REQUEST_START,
    data,
  };
};

export const sendContactRequestSuccess = (data) => {
  return {
    type: SEND_CONTACT_REQUEST_SUCCESS,
    data,
  };
};

export const sendContactRequestFailed = (error) => {
  return {
    type: SEND_CONTACT_REQUEST_FAILED,
    error,
  };
};

export const sendInviteRequest = (data) => {
  return (dispatch) => {
    dispatch(sendInviteRequestStart(data));

    let headers = {};

    if (data.cv) {
      headers["Content-Type"] = "multipart/form-data";
      data = composeFormData(data);
    }

    axios
      .post(ENDPOINT_INVITE_REQUEST, data, { headers })
      .then((response) => {
        dispatch(sendInviteRequestSuccess(response.data));
      })
      .catch((error) => {
        dispatch(sendInviteRequestFailed(error.response ? error.response.data : null));
      });
  };
};

export const sendInviteRequestStart = (data) => {
  return {
    type: SEND_INVITE_REQUEST_START,
    data,
  };
};

export const sendInviteRequestSuccess = (data) => {
  return {
    type: SEND_INVITE_REQUEST_SUCCESS,
    data,
  };
};

export const sendInviteRequestFailed = (error) => {
  return {
    type: SEND_INVITE_REQUEST_FAILED,
    error,
  };
};

export const getMediumPosts = () => {
  return (dispatch) => {
    dispatch(getMediumPostsStart());
    axios
      .get(ENDPOINT_MEDIUM)
      .then((response) => {
        dispatch(getMediumPostsSuccess(response.data));
      })
      .catch((error) => {
        dispatch(getMediumPostsFailed(error.response ? error.response.data : null));
      });
  };
};

export const getMediumPostsStart = () => {
  return {
    type: GET_MEDIUM_POSTS_START,
  };
};

export const getMediumPostsSuccess = (posts) => {
  return {
    type: GET_MEDIUM_POSTS_SUCCESS,
    posts,
  };
};

export const getMediumPostsFailed = (error) => {
  return {
    type: GET_MEDIUM_POSTS_FAILED,
    error,
  };
};

export const findReplacement = (model, id) => {
  return (dispatch) => {
    dispatch(findReplacementStart(model, id));
    axios
      .get(ENDPOINT_MIGRATE + `${model}/${id}/`)
      .then((response) => {
        dispatch(findReplacementSuccess(response.data, model, id));
      })
      .catch((error) => {
        dispatch(findReplacementFailed(error.response ? error.response.data : null, model, id));
      });
  };
};

export const findReplacementStart = (model, id) => {
  return {
    type: FIND_REPLACEMENT_START,
    model,
    id,
  };
};

export const findReplacementSuccess = (replacement, model, id) => {
  return {
    type: FIND_REPLACEMENT_SUCCESS,
    replacement,
    model,
    id,
  };
};

export const findReplacementFailed = (error, model, id) => {
  return {
    type: FIND_REPLACEMENT_FAILED,
    error,
    model,
    id,
  };
};

export const toggleRightNav = (collapsed, contentType) => {
  return {
    type: TOGGLE_RIGHT_NAV,
    collapsed,
    contentType,
  };
};
