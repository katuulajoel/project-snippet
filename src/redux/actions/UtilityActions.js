import { TOGGLE_RIGHT_NAV } from '../../configs/constants/ActionTypes';

export const toggleRightNav = (collapsed, contentType) => {
  return {
    type: TOGGLE_RIGHT_NAV,
    collapsed,
    contentType,
  };
};
