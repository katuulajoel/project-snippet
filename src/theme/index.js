import { functions } from "./functions";
import { mixins } from "./mixins";
import { variables } from "./variables";

export default {
  ...variables,
  mixins: mixins,
  functions: functions,
};
