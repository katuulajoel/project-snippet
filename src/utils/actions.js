/**
 * This get called before an async function is executed
 * @param {string} type
 * @returns
 */
export function start(type) {
  return { type };
}

/**
 * This dispatches successful Promises
 * @param {string} type
 * @param {*} data
 * @returns
 */
export function success(type, data) {
  return { type, data };
}

/**
 * For dispatching failed Promises
 * @param {string} type
 * @param {object} param1
 * @returns
 */
export function failed(type, { message }) {
  const error = message;
  return { type, error };
}
