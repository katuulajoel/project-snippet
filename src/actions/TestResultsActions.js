import React from "react";
import axios from "axios";
import { ENDPOINT_TEST_RESULTS } from "./utils/api";

import AlertDialogue from "../components/AlertDialogue";
import { openAlert } from "../components/utils/modals";
import {
  CREATE_RESULT_START,
  CREATE_RESULT_SUCCESS,
  CREATE_RESULT_FAILED,
  UPDATE_RESULT_START,
  UPDATE_RESULT_SUCCESS,
  UPDATE_RESULT_FAILED,
  FETCH_RESULT_START,
  FETCH_RESULT_SUCCESS,
  FETCH_RESULT_FAILED,
  DELETE_RESULT_START,
  DELETE_RESULT_SUCCESS,
  DELETE_RESULT_FAILED,
  // LIST_MORE_RESULTS_START,
  // LIST_MORE_RESULTS_SUCCESS,
  // LIST_MORE_RESULTS_FAILED,
  SET_FILTERS,
} from "./utils/ActionTypes";

export const createResult = (data, selectionKey) => {
  console.log(data);
  return (dispatch) => {
    dispatch(createResultStart(data, selectionKey));
    axios
      .post(ENDPOINT_TEST_RESULTS, data)
      .then((response) => {
        console.log(response);
        dispatch(createResultSuccess(response.data, selectionKey));
      })
      .catch((error) => {
        console.log(error);
        dispatch(createResultFailed(error.response ? error.response.data : null, selectionKey));
      });
  };
};

export const createResultStart = (data, selectionKey) => {
  return {
    type: CREATE_RESULT_START,
    data,
    selectionKey,
  };
};

export const createResultSuccess = (result, selectionKey) => {
  return {
    type: CREATE_RESULT_SUCCESS,
    result,
    selectionKey,
  };
};

export const createResultFailed = (error, selectionKey) => {
  if (error) {
    openAlert(<AlertDialogue msg={error.message} />, false, {
      className: "error-dailogue",
      hideActions: true,
      hideBackdrop: true,
    });
  }
  return {
    type: CREATE_RESULT_FAILED,
    error,
    selectionKey,
  };
};

export const deleteResult = (id) => {
  return (dispatch) => {
    dispatch(deleteResultStart(id));
    axios
      .delete(`${ENDPOINT_TEST_RESULTS}${id}/`, {})
      .then(() => {
        dispatch(deleteResultSuccess(id));
      })
      .catch((error) => {
        dispatch(deleteResultFailed(error.response ? error.response.data : null));
      });
  };
};

export const deleteResultStart = (id) => {
  return {
    type: DELETE_RESULT_START,
    id,
  };
};

export const deleteResultSuccess = (id) => {
  return {
    type: DELETE_RESULT_SUCCESS,
    id,
  };
};

export const deleteResultFailed = (error) => {
  return {
    type: DELETE_RESULT_FAILED,
    error,
  };
};

export const fetchResults = (selection, filter) => {
  return (dispatch) => {
    dispatch(fetchResultsStart(selection));
    axios
      .get(ENDPOINT_TEST_RESULTS, { params: filter })
      .then((response) => {
        dispatch(fetchResultsSuccess(response.data, selection));
      })
      .catch((error) => {
        dispatch(fetchResultsFailed(error.response ? error.response.data : null));
      });
  };
};

export const fetchResultsStart = (selection) => {
  return {
    type: FETCH_RESULT_START,
    selection,
  };
};

export const fetchResultsSuccess = (response, selection) => {
  return {
    type: FETCH_RESULT_SUCCESS,
    items: response.results,
    previous: response.previous,
    next: response.next,
    count: response.count,
    selection,
  };
};

export const fetchResultsFailed = (error, selection) => {
  return {
    type: FETCH_RESULT_FAILED,
    error,
    selection,
  };
};

export const updateResult = (id, result, selectionKey) => {
  return (dispatch) => {
    dispatch(updateResultStart(id, result, selectionKey));
    let headers = {};

    axios
      .patch(ENDPOINT_TEST_RESULTS + id + "/", result, {
        headers: { ...headers },
      })
      .then((response) => {
        dispatch(updateResultSuccess(response.data, id, selectionKey));
      })
      .catch((error) => {
        dispatch(updateResultFailed(error.response ? error.response.data : null, id, result, selectionKey));
      });
  };
};

export const updateResultStart = (id, result, selectionKey) => {
  return {
    type: UPDATE_RESULT_START,
    id,
    result,
    selectionKey,
  };
};

export const updateResultSuccess = (result, id, selectionKey) => {
  return {
    type: UPDATE_RESULT_SUCCESS,
    result,
    id,
    selectionKey,
  };
};

export const updateResultFailed = (error, id, result, selectionKey) => {
  return {
    type: UPDATE_RESULT_FAILED,
    error,
    result,
    id,
    selectionKey,
  };
};

export const setSelectedFilters = (filters) => {
  return {
    type: SET_FILTERS,
    filters,
  };
};
