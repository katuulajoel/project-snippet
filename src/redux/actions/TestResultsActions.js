import React from 'react';
import axios from 'axios';
import { ENDPOINT_TEST_RESULTS } from '../../utils/api';

import AlertDialogue from '../../components/AlertDialogue';
import { openAlert } from '../../utils/modals';
import {
  CREATE_RESULT_START,
  CREATE_RESULT_SUCCESS,
  CREATE_RESULT_FAILED,
  UPDATE_RESULT_START,
  UPDATE_RESULT_SUCCESS,
  UPDATE_RESULT_FAILED,
  FETCH_RESULT_START,
  FETCH_RESULT_SUCCESS,
  FETCH_RESULT_FAILED,
  DELETE_RESULT_START,
  DELETE_RESULT_SUCCESS,
  DELETE_RESULT_FAILED,
  SET_FILTERS,
} from '../../configs/constants/ActionTypes';

export const createResult = (data) => {
  return (dispatch) => {
    dispatch(createResultStart(data));
    axios
      .post(ENDPOINT_TEST_RESULTS, data)
      .then((response) => {
        dispatch(createResultSuccess(response.data));
      })
      .catch((error) => {
        dispatch(createResultFailed(error.response ? error.response.data : null));
      });
  };
};

export const createResultStart = (data) => {
  return {
    type: CREATE_RESULT_START,
    data,
  };
};

export const createResultSuccess = (result) => {
  return {
    type: CREATE_RESULT_SUCCESS,
    result,
  };
};

export const createResultFailed = (error) => {
  if (error) {
    openAlert(<AlertDialogue msg={error.message} />, false, {
      className: 'error-dailogue',
      hideActions: true,
      hideBackdrop: true,
    });
  }
  return {
    type: CREATE_RESULT_FAILED,
    error,
  };
};

export const deleteResult = (id) => {
  return (dispatch) => {
    dispatch(deleteResultStart(id));
    axios
      .delete(`${ENDPOINT_TEST_RESULTS}${id}/`, {})
      .then(() => {
        dispatch(deleteResultSuccess(id));
      })
      .catch((error) => {
        dispatch(deleteResultFailed(error.response ? error.response.data : null));
      });
  };
};

export const deleteResultStart = (id) => {
  return {
    type: DELETE_RESULT_START,
    id,
  };
};

export const deleteResultSuccess = (id) => {
  return {
    type: DELETE_RESULT_SUCCESS,
    id,
  };
};

export const deleteResultFailed = (error) => {
  return {
    type: DELETE_RESULT_FAILED,
    error,
  };
};

export const fetchResults = (filter) => {
  return (dispatch) => {
    dispatch(fetchResultsStart());
    axios
      .get(ENDPOINT_TEST_RESULTS, { params: filter })
      .then((response) => {
        dispatch(fetchResultsSuccess(response.data));
      })
      .catch((error) => {
        dispatch(fetchResultsFailed(error.response ? error.response.data : null));
      });
  };
};

export const fetchResultsStart = () => {
  return {
    type: FETCH_RESULT_START,
  };
};

export const fetchResultsSuccess = (response) => {
  return {
    type: FETCH_RESULT_SUCCESS,
    items: response.results,
    previous: response.previous,
    next: response.next,
    count: response.count,
  };
};

export const fetchResultsFailed = (error) => {
  return {
    type: FETCH_RESULT_FAILED,
    error,
  };
};

export const updateResult = (id, result) => {
  return (dispatch) => {
    dispatch(updateResultStart(id, result));
    let headers = {};

    axios
      .patch(ENDPOINT_TEST_RESULTS + id + '/', result, {
        headers: { ...headers },
      })
      .then((response) => {
        dispatch(updateResultSuccess(response.data, id));
      })
      .catch((error) => {
        dispatch(updateResultFailed(error.response ? error.response.data : null, id, result));
      });
  };
};

export const updateResultStart = (id, result) => {
  return {
    type: UPDATE_RESULT_START,
    id,
    result,
  };
};

export const updateResultSuccess = (result, id) => {
  return {
    type: UPDATE_RESULT_SUCCESS,
    result,
    id,
  };
};

export const updateResultFailed = (error, id, result) => {
  return {
    type: UPDATE_RESULT_FAILED,
    error,
    result,
    id,
  };
};

export const setSelectedFilters = (filters) => {
  return {
    type: SET_FILTERS,
    filters,
  };
};
