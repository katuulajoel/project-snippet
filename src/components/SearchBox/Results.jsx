/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import PropTypes from "prop-types";
import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import styled from "styled-components";

/* -------------------------- Internel Dependencies ------------------------- */
import Progress from "../Progress";
import LoadMore from "../LoadMore";
import { listMoreInvoices } from "../../redux/actions/InvoiceActions";
import SummaryPlaceholder from "../SummaryPlaceholder/SummaryPlaceholder";

const Results = ({ searchTerm, clearSearch, navHieght }) => {
  const { search, isMakingRequest } = useSelector(({ Invoice }) => Invoice);
  const dispatch = useDispatch();
  return (
    <>
      {searchTerm && (
        <Wrapper navHieght={navHieght?.current?.offsetHeight}>
          {isMakingRequest.search && <Progress />}
          {search.data.length == 0 ? (
            <SummaryPlaceholder description="No results found" />
          ) : (
            <div className="results">
              {search.data.length && (
                <span className="counter" data-testid="counter">
                  Showing <strong>{search.data.length}</strong> results for{" "}
                  <strong>&quot;{searchTerm}&quot;</strong>
                </span>
              )}

              {isMakingRequest.search ? null : search.data.length ? (
                <div className="section">
                  <div className="title">Payments</div>
                  {search.data.map((invoice) => {
                    return (
                      <Link
                        key={invoice?.id}
                        to={`/projects/${invoice?.project.id}/pay`}
                        className="result-item"
                        onClick={clearSearch}
                      >
                        <b>{invoice?.project.title}</b>: {invoice?.title}
                      </Link>
                    );
                  })}

                  <LoadMore
                    type="text"
                    hasMore={search.next && search.next != ""}
                    isLoadingMore={isMakingRequest.search}
                    onLoadMore={() =>
                      dispatch(listMoreInvoices(search.next, true))
                    }
                    variant="outline-primary"
                  >
                    View More
                  </LoadMore>
                </div>
              ) : null}
            </div>
          )}
        </Wrapper>
      )}
    </>
  );
};

const Wrapper = styled.div`
  width: 413px;
  top: ${(props) => props.navHieght}px;
  height: calc(100vh - ${(props) => props.navHieght + "px"});
  right: 0;
  overflow-y: scroll;

  ${(props) => props?.theme?.mixins?.scrollbar}
  padding: 40px;
  position: absolute;
  background: #fff;
  border: 1px solid #efefef;
  margin-top: 7.0625rem;

  .results {
    font-size: 16px;
    line-height: 19px;
    color: #151a30;

    .counter {
      b {
        font-weight: 600;
      }
    }
    .section {
      padding: 40px 0;
      border-bottom: 1px solid #edf1f7;

      &:last-of-type {
        border-bottom: none;
      }

      .title {
        font-weight: 600;
        margin-bottom: 16px;
      }
      a {
        display: block;
        text-decoration: none;
        font-weight: 500;
        margin-bottom: 16px;
        color: #151a30;
        b {
          font-weight: 600;
        }
      }

      .text-link {
        button {
          border: none;
          padding: 0;
          font-weight: 600;
          font-size: 14px;
          line-height: 17px;
          color: #da3451;
        }
        button:hover,
        button:focus {
          background-color: #fff;
          color: #da3451;
        }
      }

      .avatar.avatar-dash {
        width: 48px;
        height: 48px;

        &.avatar-initials {
          font-size: 16px;
        }
      }
    }
  }
`;

Results.propTypes = {
  searchTerm: PropTypes.string,
  clearSearch: PropTypes.func,
  navHieght: PropTypes.any,
};

export default Results;
