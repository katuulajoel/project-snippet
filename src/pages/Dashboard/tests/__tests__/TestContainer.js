import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';

import TestContainer from '../TestContainer';
import { mount } from 'enzyme/build';
import Results, { dummyResults } from './Results';
import * as actions from '../../../../redux/actions/TestResultsActions';
import { toggleRightNav } from '../../../../redux/actions/UtilityActions';
import theme from '../../../../assets/theme';

const middlewares = [thunk];

const mockAppState = {
  Auth: {
    user: { uid: 123, email: 'test@gmail.com' },
    isMakingRequest: {},
  },
  Invoice: {
    isMakingRequest: {},
    errors: {},
    summary: {},
    list: { data: [], count: 0, next: '', previous: '' },
    invoice: {},
    csv: {},
  },
  TestResults: {
    count: {},
    errors: {},
    isFetching: { default: true },
    isSaved: {},
    isSaving: {},
    next: {},
    previous: {},
    results: [],
    selectedFilters: [],
  },
};

const count = dummyResults.length;
const props = {
  testResults: { results: dummyResults },
  count,
};

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state || mockAppState);
};

describe('Test list container', () => {
  it('Should match snapshot test', () => {
    const { asFragment } = render(
      <Provider store={mockAppStore()}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <TestContainer collapseRightNav={toggleRightNav} TestResults={mockAppState.TestResults} />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
    expect(
      asFragment(<TestContainer collapseRightNav={toggleRightNav} TestResults={mockAppState.TestResults} />)
    ).toMatchSnapshot();
  });

  it('should render children', async () => {
    const store = mockAppStore();
    mount(
      <BrowserRouter>
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <TestContainer collapseRightNav={toggleRightNav} TestResults={mockAppState.TestResults}>
              <></>
            </TestContainer>
          </ThemeProvider>
        </Provider>
      </BrowserRouter>
    );

    const expectedActions = [{ type: 'FETCH_RESULT_START' }];

    expect(store.getActions()).toMatchObject(expectedActions);
  });

  it('should show progress component', async () => {
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={mockAppStore()}>
          <ThemeProvider theme={theme}>
            <TestContainer collapseRightNav={toggleRightNav} TestResults={mockAppState.TestResults}>
              <></>
            </TestContainer>
          </ThemeProvider>
        </Provider>
      </BrowserRouter>
    );

    expect(wrapper.find('Progress').exists()).toBeTruthy();
  });

  it('should load data on page change', async () => {
    const testResultsActions = jest.spyOn(actions, 'fetchResults');
    const store = mockAppStore({
      ...mockAppState,
      Invoice: {
        list: {
          data: [dummyResults],
          count: dummyResults.length,
        },
        isMakingRequest: {},
      },
      TestResults: {
        count: {},
        errors: { fetch: null },
        isFetching: { default: false },
        isSaved: {},
        isSaving: {},
        next: {},
        previous: {},
        results: [],
        selectedFilters: [],
      },
    });
    // eslint-disable-next-line no-unused-vars
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <TestContainer collapseRightNav={toggleRightNav} TestResults={mockAppState.TestResults}>
              <></>
            </TestContainer>
          </ThemeProvider>
        </Provider>
      </BrowserRouter>
    );

    // const nextPage = wrapper.find("li.next");
    // nextPage.simulate("click");

    expect(testResultsActions).toHaveBeenCalled();
    // expect(testResultsActions).toBeCalledWith({
    //   "6LMlpGnA": { page_size: 20 },
    //   //  2: "6LMlpGnA", {"page_size": 20}
    // });
  });

  it('load new results on change to new status page', () => {
    const fetchResultsStub = jest.spyOn(actions, 'fetchResults');

    const store = mockAppStore({
      Invoice: {
        list: {
          data: [],
          count: 0,
        },
        isMakingRequest: {},
      },

      TestResults: {
        count: {},
        errors: {},
        isFetching: { default: true },
        isSaved: {},
        isSaving: {},
        next: {},
        previous: {},
        results: dummyResults,
        selectedFilters: [],
        isMakingRequest: {},
      },
    });
    // eslint-disable-next-line no-unused-vars
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <TestContainer collapseRightNav={toggleRightNav} TestResults={mockAppState.TestResults}>
              <Results {...props} />
            </TestContainer>
          </ThemeProvider>
        </Provider>
      </BrowserRouter>
    );
    expect(fetchResultsStub).toHaveBeenCalled();
    expect(fetchResultsStub).toHaveBeenCalledTimes(2);

    const h3 = container.querySelector('h3'); 
    expect(h3).toBeTruthy();
    expect(h3.innerHTML).toBe('Results');

    // container.setProps({ match: { params: { filter: "overdue" } } });
  });
});
