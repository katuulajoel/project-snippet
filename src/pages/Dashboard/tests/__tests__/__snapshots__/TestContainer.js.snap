// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard test - Test Results Should match snapshot test 1`] = `
Array [
  <div
    className="table-responsive"
  >
    <table
      className="sc-kEqXSa sc-crzoAE bukXYq eTSFPk table"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            <div>
              User
              <div
                className="caret-group"
              >
                
              </div>
            </div>
          </th>
          <th
            colSpan={1}
            role="columnheader"
            style={
              Object {
                "cursor": undefined,
              }
            }
          >
            <div>
              Coding Tests
              <div
                className="caret-group"
              >
                
              </div>
            </div>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            <div>
              Comms Check
              <div
                className="caret-group"
              >
                
              </div>
            </div>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            <div>
              MBTI Profile
              <div
                className="caret-group"
              >
                
              </div>
            </div>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            <div>
              IQ Tests
              <div
                className="caret-group"
              >
                
              </div>
            </div>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            <div>
              SA Tests
              <div
                className="caret-group"
              >
                
              </div>
            </div>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            <div>
              C. Of Conduct
              <div
                className="caret-group"
              >
                
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            role="cell"
          >
            <div>
              <div
                className="sc-iCoGMd daNtNP avatar-dash avatar-initials"
              >
                <div
                  className="avatar avatar-dash avatar-initials"
                  style={Object {}}
                >
                  WD
                </div>
              </div>
              Wd Dw
              <div
                className="edit-action"
              >
                <button
                  className="btn btn-icon sc-hKFxyN hwaQcw btn-edit  "
                  form={null}
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="tg-ic-circle-edit-outline tunga-ic-sz-main "
                  />
                </button>
              </div>
            </div>
          </th>
          <td
            role="cell"
          >
            <div
              className="result"
            >
              <span
                className="passed"
              >
                react
                 - 
                100
              </span>
            </div>
          </td>
          <td
            role="cell"
          >
            <div
              className="result"
            >
              <span
                className="passed"
              >
                Very good
              </span>
            </div>
          </td>
          <td
            role="cell"
          >
            <div
              className="result"
            >
              <span
                className="none"
                style={
                  Object {
                    "textTransform": "uppercase",
                  }
                }
              >
                entj
              </span>
            </div>
          </td>
          <td
            role="cell"
          >
            <div
              className="result"
            >
              <span
                className="failed"
              >
                100
              </span>
            </div>
          </td>
          <td
            role="cell"
          >
            <div
              className="result"
            >
              <span
                className="passed"
              >
                100
              </span>
            </div>
          </td>
          <td
            role="cell"
          >
            <div
              className="result"
            >
              <span
                className="passed"
              >
                100
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="sc-dIsUp xUNzk"
  >
    <span>
      Showing
    </span>
    <div
      className="sc-bqGGPW bZBMeg"
    >
      <select
        className="form-control form-control "
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          -- Select --
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value="100"
        >
          100
        </option>
      </select>
      <i
        className="tg-ic-rounded-keyboard-arrow-down tunga-ic-sz-sm "
      />
    </div>
    <span
      style={
        Object {
          "marginLeft": "16px",
        }
      }
    >
      of 
      1
    </span>
    <nav
      aria-label="Page navigation example"
      className=""
    >
      <ul
        className="pagination"
      >
        <li
          className="sc-gKAaRy BYywQ page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#"
          >
            <i
              className="tg-ic-arrow-left tunga-ic-sz-sm "
            />
          </a>
        </li>
        <div
          className="Table__visiblePagesWrapper"
          style={
            Object {
              "display": "flex",
            }
          }
        />
        <li
          className="sc-gKAaRy BYywQ next page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#"
          >
            <i
              className="tg-ic-arrow-right tunga-ic-sz-sm "
            />
          </a>
        </li>
      </ul>
    </nav>
  </div>,
]
`;
